pallets/compliance-manager/src/lib.rs

```diff
@@ -336,7 +337,9 @@ decl_module! {
             ensure!(<Identity<T>>::is_identity_exists(&issuer.issuer), Error::<T>::DidNotExist);
             TrustedClaimIssuer::try_mutate(ticker, |issuers| {
                 ensure!(!issuers.contains(&issuer), Error::<T>::IncorrectOperationOnTrustedIssuer);
-                Self::base_verify_compliance_complexity(&AssetCompliances::get(ticker).requirements, issuers.len() + 1)?;
+                let def_complexity = trusted_issuers_complexity(&issuers)
+                    .saturating_add(issuer.complexity());
```

Comment:
The complexity we are measuring for compliance rules is the worse case number of reads. IIRC the issuer complexity does not affect the number of reads. A more complex issuer will require slightly more time to filter but that should be negligible when compared to complexity of a read operation. Therefore, IMO we should only care about the number of issuers, not their complexity.